src/main/kotlin/org/misarch/discount/graphql/Query.kt

```diff
@@ -50,13 +50,13 @@ class Query(
         return dfe.getDataLoader<UUID, Coupon>(CouponDataLoader::class.simpleName!!).load(id)
     }
 
-    @GraphQLDescription("Find all applicable discounts for a product variant and user, and list of coupon ids.")
+    @GraphQLDescription("Find all applicable discounts for user and a list of product variant, count and coupon id triples")
     @InaccessibleDirective
     suspend fun findApplicableDiscounts(
         @GraphQLDescription("The input for the findApplicableDiscounts query.")
         input: FindApplicableDiscountsInput
-    ): List<Discount> {
-        return discountService.findApplicableDiscounts(input).map { it.toDTO() }
+    ): List<List<Discount>> {
+        return discountService.findApplicableDiscounts(input).map { discounts -> discounts.map { it.toDTO() } }
```

Comment:
Wait, why a `List<List<>>`?
That sounds really strange to consume and understand…
At best document what this return value means and how to interpret it.
Or switch it back to a `List<Discount>`

Reply:
We only want one request for multiple order items, each order items has a collection of coupons, therefore i think this is this mapping. Even through @nk-coding said we do not want to make this mapping index based? But i guess this should do the job.

Reply:
see https://github.com/MiSArch/discount/pull/3#discussion_r1527106849
but still I changed it a bit with ec40bffa46518ffee4b516196f78921f508238f1